### PR TITLE
Create admin schedule review table

### LINKED ISSUE #156: Schedule review table row
### Summary and Background
Schedule review table row outlines basic information for a shift and one of its volunteers.

### User Story
As an Admin, I want to be able see who is volunteering at what time.
As an Admin, I want to be able to remove a volunteer if I don't want them to volunteer at a shift.

### Solution (Core Elements)
Create a table row for the Schedule review table. There are three row types based on shifts available:

For a filled shift (first image),
Left: Start and end times of the shift with total length
Middle: Volunteer assigned
Right: Remove volunteer button

When no shifts are available (second image), just indicate that no shifts are available.

When there is a shift with no volunteers assigned (third image),
Left: Start and end times of the shift with total length.
Middle: "No volunteers"

### Mockups

Other designs:
- https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=888%3A8374
- https://www.figma.com/file/KWaJRNJcOiW8ThZJ8Sl6On/Volunteer-Experience?node-id=157%3A3123.

**Extremely similar to**: https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/issues/uwblueprint/sistering/122

### Technical
Create a component called `AdminScheduleTableRow.tsx` in `frontend\src\components\admin\schedule`. It should take the volunteer name and posting times as mandatory props. 

## Dev Notes
* Chakra [table](https://chakra-ui.com/docs/data-display/table)

### LINKED ISSUE #158: Schedule review table
### Summary and Background
Schedule review table holds rows for assigned volunteers in a week.

### User Story
As an Admin, I want to be able see all assigned volunteers in a clean and concise manner.
As an Admin, I want to be able to remove any assigned volunteer if I want.

### Solution (Core Elements)
Create Schedule review table. Each day in a week is displayed, and all volunteers will appear under its date, grouped by shift time. If a posting has no shifts in a day, a no shifts row will appear.

### Mockups

Other designs:
- https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=888%3A8374
- https://www.figma.com/file/KWaJRNJcOiW8ThZJ8Sl6On/Volunteer-Experience?node-id=157%3A3123.

Extremely similar to: https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/issues/uwblueprint/sistering/120

### Technical
* This ticket does **not** include the table header with week selectors
* This ticket does **not** include data fetching, please use some mock volunteer signups data to test
* Create a component in AdminScheduleTable.tsx in frontend/src/components/admin/schedule, rendering the row components created in #156 in a table
* The component takes a postingId as a prop
* The component should maintain currentWeek and signups as a state variables

## Dev Notes
* Chakra [table](https://chakra-ui.com/docs/data-display/table)

### PR DESCRIPTION
**Pull request by lambo-liu**
## Ticket link

Closes #156 
Closes #158 

## Implementation description
* Implemented the **schedule review table** and its table rows, which display all volunteer shifts of a certain week.
* Note that this table implementation is only for displaying the shifts in one week (passed in via props), and does not include the date navigation header.

## Steps to test
1. Add the following code to `Default.tsx`, importing components/variables when necessary:
```
<Container maxW="container.xl">
  <AdminScheduleTable schedule={TableTestData} />
</Container>
```
2. Check http://localhost:3000/ to see that `TableTestData` is rendered correctly. The test data includes three types of table rows, as specified in #156.

## What should reviewers focus on?
* Clean React code, correct separation of concern for the props & states.
* Stay hydrated 💧 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by LenaNguyen — COMMENTED** [review]
padding for the rows is a little off 
image

I'm pretty sure the horizontal padding should be equal so let's use 47px. 
Since the Td elements have their own padding and it would be annoying to remove their padding each time, just add some extra padding to `tr`. i.e. px="20px", py="2.5px"

**Review by LenaNguyen — COMMENTED** [review]
Also for UI changes can you pls add a demo on the default page or something next time 🙏 my docker takes a million years to start so sometimes it's much faster to use the firebase preview. The downside is you'd have to remove it after get another approval tho.

**Comment by LenaNguyen** [comment]
> padding for the rows is a little off image
> 
> I'm pretty sure the horizontal padding should be equal so let's use 47px. Since the Td elements have their own padding and it would be annoying to remove their padding each time, just add some extra padding to `tr`. i.e. px="20px", py="2.5px"

Actually idk if this is that important @sherryhli . If we change the theme to accommodate for this padding then other places where we use a `Tr` but isn't formatted like this will need to be changed.  The default spacing from Chakra also looks fine.

File: frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      <Td>
+        {!volunteer && !userId ? (
+          <Text color="text.gray">No volunteers</Text>
+        ) : (
+          <Text>{volunteer}</Text>
+        )}
+      </Td>
+      <Td textAlign="right">
+        {volunteer && userId && (
+          <Button color="text.critical" variant="none">
```
Comment (sherryhli) [inline frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx:39]: Hmm I'm seeing different y-paddings between the 3 types of rows, they should equal according to the designs

image

**Review by sherryhli — CHANGES_REQUESTED** [review]
Nice! 🔥 Some small code improvement suggestions and a minor UI deviation, LGTM otherwise!

**Comment by sherryhli** [comment]
> > padding for the rows is a little off image
> > I'm pretty sure the horizontal padding should be equal so let's use 47px. Since the Td elements have their own padding and it would be annoying to remove their padding each time, just add some extra padding to `tr`. i.e. px="20px", py="2.5px"
> 
> Actually idk if this is that important @sherryhli . If we change the theme to accommodate for this padding then other places where we use a `Tr` but isn't formatted like this will need to be changed. The default spacing from Chakra also looks fine.

@LenaNguyen Yeah I'm for staying consistent to a single theme variant if possible, could we check with design to see if this small deviation is okay?

**Comment by sherryhli** [comment]
Just realized something: with the way we plan on fetching data for the schedule creation frontend (running a query on the signups table rather than shifts), we wouldn't know about any shifts without signups. That means the row type below would not be seen, and the shift would also not be shown as a dot in the calendar.

row

cc @JoshuaYe @LenaNguyen Do you have any input on the necessity and priority of this feature? How useful is it to view shifts that do not have any signups in schedule creation?

(@lambo-liu no change needed for now, please keep this row type in the code)

**Comment by LenaNguyen** [comment]
> Just realized something: with the way we plan on fetching data for the schedule creation frontend (running a query on the signups table rather than shifts), we wouldn't know about any shifts without signups. That means the row type below would not be seen, and the shift would also not be shown as a dot in the calendar.
> 
> row
> 
> cc @JoshuaYe @LenaNguyen Do you have any input on the necessity and priority of this feature? How useful is it to view shifts that do not have any signups in schedule creation?
> 
> (@lambo-liu no change needed for now, please keep this row type in the code)

It seems useful to know that there are some shifts that no one is registered for in the overview

File: frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx (hunk 1 above)
Comment (lambo-liu) [reply frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx:39]: Thanks for catching this 👍

**Review by LenaNguyen — APPROVED** [review]
LGTM!

**Review by sherryhli — APPROVED** [review]

**Pull request merged**